hermit-sys/src/net/mod.rs

```diff
@@ -103,6 +111,50 @@ where
 		{
 			// just to make progress
 		}
+		#[cfg(feature = "dhcpv4")]
+		let config = self
+			.dhcp
+			.poll(&mut self.iface, &mut self.sockets, timestamp)
+			.unwrap_or_else(|e| {
+				debug!("DHCP: {:?}", e);
+				None
+			});
+		#[cfg(feature = "dhcpv4")]
+		config.map(|config| {
+			debug!("DHCP config: {:?}", config);
+			if let Some(cidr) = config.address {
+				if cidr != self.prev_cidr {
+					self.iface.update_ip_addrs(|addrs| {
+						addrs.iter_mut().next().map(|addr| {
+							*addr = IpCidr::Ipv4(cidr);
+						});
+					});
+					self.prev_cidr = cidr;
+					info!("Assigned a new IPv4 address: {}", cidr);
+				}
+			}
+
+			config.router.map(|router| {
+				self.iface
+					.routes_mut()
+					.add_default_ipv4_route(router)
+					.unwrap()
```

Comment:
Are you unwrapping because you want to panic if setting the route fails, or because you are sure that this can never fail in this context?

Reply:
I want to panic.